packages/wonder-blocks-announcer/src/index.ts

```diff
@@ -0,0 +1,334 @@
+// TODO: publish wonder-blocks-style package WB-1776
+// import {srOnly} from "../../wonder-blocks-style/src/styles/a11y";
+import type {
+    PolitenessLevel,
+    RegionDef,
+    RegionDictionary,
+    RegionFactory,
+} from "./announcer.types";
+
+const TIMEOUT_DELAY = 5000;
+let announcer: Announcer | null = null;
+
+export type SendMessageProps = {
+    message: string;
+    level?: PolitenessLevel;
+    removalDelay?: number;
+};
+
+/**
+ * Public API method to send screen reader messages.
+ * @param {string} message The message to send.
+ * @param {PolitenessLevel} level Polite or assertive announcements
+ * @param {number} removalDelay Optional duration to remove a message after sending. Defaults to 5000ms.
+ * @returns {string} IDREF for targeted live region element
+ */
+export function sendMessage({
+    message,
+    level = "polite", // TODO: decide whether to allow other roles, i.e. role=`timer`
+    removalDelay,
+}: SendMessageProps): string {
+    announcer = Announcer.getInstance();
+
+    if (typeof jest === "undefined") {
+        setTimeout(() => {
+            return announcer?.announce(message, level, removalDelay);
+        }, 100);
+    } else {
+        // If we are in a test environment, announce without waiting
+        return announcer.announce(message, level, removalDelay);
+    }
+    return "";
+}
+/**
+ * Public API method to clear screen reader messages after sending.
+ * Clears all regions by default.
+ * @param {string} id Optional id of live region element to clear.
+ */
+export function clearMessages(id?: string) {
+    if (id && document?.getElementById(id)) {
+        announcer?.clear(id);
+    } else if (document) {
+        announcer?.clear();
+    }
+}
+
+/**
+ * Internal class to manage screen reader announcements.
+ */
+class Announcer {
+    private static _instance: Announcer;
+    node: HTMLElement | null = null;
+    regionFactory: RegionFactory = {
+        count: 2,
+        aIndex: 0,
+        pIndex: 0,
+    };
+    dictionary: RegionDictionary = new Map();
+
+    private constructor() {
+        if (typeof document !== "undefined") {
+            const topLevelId = `wbAnnounce`;
```

Comment:
**suggestion:** This is a good candidate for a constant at the file level so this ID can be shared across the file. wdyt?

Reply:
That's what I had at first, but it added cognitive load when working with the API. It makes the file a lot more scannable to locate this ID closer to the code it's used on!

Reply:
fair enough!